src/PointwiseFunctions/InitialDataUtilities/InitialData.hpp

```diff
@@ -18,7 +18,7 @@ namespace initial_data {
  * \brief The abstract base class for initial data of evolution systems. All
  * analytic solutions and analytic data must virtually inherit from this class.
  */
-class InitialData : public PUP::able {
+class InitialData : public virtual PUP::able {
```

Comment:
This seems like a really bad idea. Who's responsible for calling the base class `pup` function when you have multiple in the hierarchy? I don't think calling the function more than once is a good idea (seems like asking for UB). I'm really strongly opposed to this change unless there is clear documentation from Charm++ that this is supported and how to handle it correctly. This is not something I think we should leave to "the tests seem to pass so it must be fine". It could result in horrendous runtime errors.